src/fromager/constraints.py

```diff
@@ -34,12 +37,18 @@ def add_constraint(self, unparsed: str) -> None:
             self._data[canon_name] = req
 
     def load_constraints_file(self, constraints_file: str | pathlib.Path) -> None:
-        """Load constraints from a constraints file"""
+        """Load constraints from a constraints file or URI"""
         logger.info("loading constraints from %s", constraints_file)
         content = requirements_file.parse_requirements_file(constraints_file)
         for line in content:
             self.add_constraint(line)
 
+    def dump_constraints_file(self, constraints_file: pathlib.Path) -> None:
+        """Dump all constraints into a file"""
+        with constraints_file.open("w", encoding="utf-8") as f:
+            for req in self._data.values():
+                f.write(f"{req}\n")
```

Comment:
Do we log the constraints as we parse the input? I know we log matching constraints as we process individual packages. It may also be useful to log the contents of this file to help with debugging.